src/prefect/tasks/secrets/vault_secret.py

```diff
@@ -0,0 +1,161 @@
+            raise PermissionError(
+                "Unable to autheticate with vault using supplied credentials"
+            )
+        self.logger.debug("passed vault authentication check")
+        # regex to parse path into 2 named parts: <mount_point>/<path>
+        secret_path_re = r"^(?P<mount_point>[^/]+)/(?P<path>.+)$"
+        m = re.fullmatch(secret_path_re, name)
+        if m is None:
+            raise KeyError(
+                f'Invalid secret path: {name}.  Expected: "<mount_point>/<path>"'
+            )
+        vault_path = m.groupdict()
+        value = ""
+        try:
+            vault_secret = client.secrets.kv.v2.read_secret_version(
+                path=vault_path["path"], mount_point=vault_path["mount_point"]
+            )
+            value = vault_secret["data"]["data"]
+        except hvac.exceptions.InvalidPath as exc:
+            raise KeyError(f"Secret not found: {vault_path['path']}") from exc
+        except hvac.exceptions.Forbidden as exc:
+            raise PermissionError(f"Access forbidden: {vault_path['path']}") from exc
+        return value
+
+    @defaults_from_attrs("name")
+    def run(self, name: str = None):
+        """
+        The run method for VaultSecret Task.  Note that this method first checks context
+        for the local secret value, and if not found either raises an error or queries
+        the Vault service, depending on whether `config.cloud.use_local_secrets` is `True`
+        or `False`.
+
+        Args:
+            - name (str, optional): The secret name defined by the Vault secret
+                path "<mount-point>/<path>".  Defaults to the name provided at initialization.
+
+        Returns:
+            - Any: the underlying value of the Vault Secret
+
+        Raises:
+            - ValueError: if a `result` keyword is passed, or if called within a flow building context,
+                or `use_local_secrets=True` and the Secret does not exist
+            - KeyError: if `use_local_secrets=False` and the Client fails to retrieve the secret
+            - PermissionError: if `use_local_secrets=False` and the Client is not authorised to read
+                the secret
+            - ClientError: if unable to find VAULT_ADDR or vault_addr environment variable
+        """
+        if isinstance(prefect.context.get("flow"), prefect.core.flow.Flow):
+            raise ValueError(
+                "Secrets should only be retrieved during a Flow run, "
+                "not while building a Flow."
+            )
+
+        secrets = prefect.context.get("secrets", {})
+        value = None
+        try:
+            value = secrets[name]
+        except KeyError:
+            if prefect.context.config.cloud.use_local_secrets is False:
+                # lookup the secret value in Vault
+                value = self._get_vault_secret(name)
+            else:
+                raise ValueError(f'Local secret "{name}" not found.') from None
```

Comment:
I'm not sure this should re-use the Prefect local secret logic but I see the use-case for it. @cicdw your opinion would be helpful here.

Reply:
Yea I don't think we should reuse the local secrets logic here honestly; that flag is intended for Prefect Cloud secrets specifically and no other custom Secret implementation uses it so that would introduce non-uniform behavior

Reply:
I'll refactor to remove the use of local secrets.
I'd originally taken the approach that a secret was a secret and it could be accessed in different remote services.  The local storage of a secret then took precedence and could be used to support isolated/local development.
In practice, we seldom use the local secrets option over accessing a running Vault service, especially given the fact that secrets are usually needed to interact with other remote services anyway.

Reply:
local secret lookup removed.  VaultSecret now only retrieves secret values from the vault service.